### ISSUE TITLE
A possible indentation_linter false-positive for styler and then lintr workflow

### ISSUE THREAD

**Issue by jeksterslab** [r-lib/lintr#2036]
I use `styler` (1.10.1) and then `lintr` (3.1.0) to style R code first before linting. In the toy example below, the `original` code gets `styled with styler` like below. The styled code, however, gets flagged by the latest release of `lintr`.

`original`

```r
foo <- function(pkg = c("styler", "lintr", "usethis",
  "tidyverse", "ggplot2", "rmarkdown", "knitr")) {
  print(pkg)
}
```

`styled with styler`

```r
foo <- function(pkg = c(
                  "styler", "lintr", "usethis",
                  "tidyverse", "ggplot2", "rmarkdown", "knitr"
                )) {
  print(pkg)
}
```

**Comment by MichaelChirico** [comment]
FWIW I would encourage double indentation here:

```r
foo <- function(
    pkg = c(...)) {
}
```

**Comment by mjsteinbaugh** [comment]
I'm seeing similar false positives as well:

https://github.com/acidgenomics/r-acidgsea/blob/v0.8.8/R/FGSEAList-methods.R#L39
https://github.com/acidgenomics/r-acidgsea/blob/v0.8.8/R/enrichedGeneSets-methods.R#L55

```r
`FGSEAList,RankedList` <- # nolint
    function(object,
             geneSetFiles,
             BPPARAM = BiocParallel::bpparam() # nolint
    ) {
```

```r
.enrichedGeneSets <-
    function(object,
             alphaThreshold,
             nesThreshold,
             direction,
             idCol, # pathway
             alphaCol, # padj
             nesCol # NES
    ) {
```

Best,
Mike

**Comment by mjsteinbaugh** [comment]
Here's another example:

https://github.com/acidgenomics/r-bcbiosinglecell/blob/develop/R/AllGenerators.R#L33

```r
bcbioSingleCell <-
    function(uploadDir,
             sampleMetadataFile = NULL,
             organism = NULL,
             ensemblRelease = NULL,
             genomeBuild = NULL,
             gffFile = NULL,
             transgeneNames = NULL,
             interestingGroups = "sampleName",
             BPPARAM = BiocParallel::bpparam() # nolint
    ) {
```

**Comment by AshesITR** [comment]
@mjsteinbaugh Your code confirms to `hanging_indent_style = "always"`. The tidyverse styleguide mandates a block indent for your arrangement of parameters.
``` r
lintr::lint(
  r"(bcbioSingleCell <-
    function(uploadDir,
             sampleMetadataFile = NULL,
             organism = NULL,
             ensemblRelease = NULL,
             genomeBuild = NULL,
             gffFile = NULL,
             transgeneNames = NULL,
             interestingGroups = "sampleName",
             BPPARAM = BiocParallel::bpparam()
    ) {
    }
)", linters = lintr::indentation_linter(indent = 4L, hanging_indent_style = "always"))
```

<sup>Created on 2023-08-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by AshesITR** [comment]
@jeksterslab A common ground for your example code, where `styler` doesn't change the code and `lintr` doesn't lint, is this:

```r
foo <- function(
    pkg = c(
      "styler", "lintr", "usethis",
      "tidyverse", "ggplot2", "rmarkdown", "knitr"
    )) {
  print(pkg)
}
```

**Comment by mjsteinbaugh** [comment]
Ah thanks @AshesITR ! I'll take a look into that

**Comment by mjsteinbaugh** [comment]
@AshesITR Figured it out, it was the single `# nolint` comment inside the function block that was freaking out the settings.
